.github/workflows/build_and_test_cmake.yml

```diff
@@ -0,0 +1,441 @@
+name: Build and test w/ CMake
+on: [push, pull_request]
+
+env:
+  BUILD_DIR: Dist
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v3
+      - uses: pre-commit/action@v2.0.3
+
+  linux:
+    runs-on: ubuntu-20.04
+    container: seladb/${{ matrix.image }}
+    strategy:
+      matrix:
+        include:
+          - image: ubuntu2204
+            python: python3
+            config-zstd: OFF
+          - image: ubuntu2004
+            python: python3
+            config-zstd: OFF
+          - image: ubuntu2004-zstd
+            python: python3
+            config-zstd: ON
```

Comment:
Maybe we can keep Zstd tests in CirrusCI (both for Linux and macOS). The reason we use multiple CI systems is to parallelize more and reduce the overall build time (which is too high even today I think...)

Reply:
With my little experience github CI seems quite fast even with both legacy and cmake build system.

If FreeBSD was supported on Github CI, I think I would have moved everything on Github CI as it's better integrated and jobs doesn't never wait a lot before being runned unlike on CirrusCI

Reply:
Yes, I agree that GitHub Actions is quite fast, however its free tier has limits:
https://docs.github.com/en/actions/learn-github-actions/usage-limits-billing-and-administration#usage-limits

Right now we have more than 20 jobs, which means some of the jobs need to wait for others to finish.
We anyway have CirrusCI for FreeBSD and in my experience it's also quite fast, reliable and easy to integrate so I think we can keep some of the jobs there.